### PR TITLE
Schema migrations for user data

### PR DESCRIPTION
**Pull request by grigoryk**

### CODE REVIEW

File: app/src/test/java/org/mozilla/firefox/vpn/user/data/DataMigrationTest.kt [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+import org.junit.runner.RunWith
+import org.mozilla.firefox.vpn.WORKING_SDK
+import org.mozilla.firefox.vpn.service.Subscription
+import org.mozilla.firefox.vpn.service.User
+import org.mozilla.firefox.vpn.service.VpnInfo
+import org.robolectric.annotation.Config
+
+@RunWith(AndroidJUnit4::class)
+@Config(sdk = [WORKING_SDK])
+class DataMigrationTest {
```
Comment (rlr) [inline app/src/test/java/org/mozilla/firefox/vpn/user/data/DataMigrationTest.kt:22]: 🙌 for tests

File: app/src/main/java/org/mozilla/firefox/vpn/user/data/Schema.kt [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        val rawUserInfo = prefs.getString(PREF_USER_INFO, null)
+        val rawAuthToken = prefs.getString(PREF_AUTH_TOKEN, null)
+        // If there's no user data, we're just operating with current schema.
+        if (rawUserInfo == null) {
+            return codeVersion
+        }
+
+        // If there's a 'token' present in user data, assume we're at version 1.
+        if (rawUserInfo.contains("token")) {
+            return 1
```
Comment (rlr) [inline app/src/main/java/org/mozilla/firefox/vpn/user/data/Schema.kt:82]: nice

**Comment by grigoryk** [comment]
Tested the following scenarios:
- fresh install
  - everything works as expected (able to login, etc)
- first-time upgrade from pre 1.1 version
  - app continues to be logged in
- upgrade from pre 1.1 version that was already upgraded with a schema change (and got logged out)
  - on first start after an upgrade, app is now again logged-in!

**Review by rlr — APPROVED** [review]
lets do it!

**Comment by rlr** [comment]
looks like lint failed https://app.circleci.com/pipelines/github/mozilla-mobile/guardian-vpn-android/1195/workflows/46cfac2c-81d0-4526-93ba-08133d4e5ae6/jobs/1387

**Pull request merged**